zcash_client_backend/src/fees/common.rs

```diff
@@ -11,16 +14,131 @@ use zcash_primitives::{
 };
 use zcash_protocol::ShieldedProtocol;
 
-use crate::data_api::WalletMeta;
+use crate::data_api::{InputSource, WalletMeta};
 
 use super::{
-    sapling as sapling_fees, ChangeError, ChangeValue, DustAction, DustOutputPolicy,
-    EphemeralBalance, SplitPolicy, TransactionBalance,
+    sapling as sapling_fees, ChangeError, ChangeStrategy, ChangeValue, DustAction,
+    DustOutputPolicy, EphemeralBalance, SplitPolicy, TransactionBalance,
 };
 
 #[cfg(feature = "orchard")]
 use super::orchard as orchard_fees;
 
+/// A change strategy that attempts to split the change value into some number of equal-sized notes
+/// as dictated by the included [`SplitPolicy`] value.
+pub struct CommonChangeStrategy<I, FR> {
```

Comment:
```suggestion
pub struct SinglePoolChangeStrategy<I, FR> {
```
And related changes elsewhere